tracer/src/Datadog.Trace/Debugger/RateLimiting/ProbeRateLimiter.cs

```diff
@@ -55,6 +55,11 @@ public bool Sample(string probeId)
 
         public void SetRate(string probeId, int samplesPerSecond)
         {
+            if (_samplers.TryGetValue(probeId, out _))
```

Comment:
Does it really make sense not to do anything here? What if the sampling rate changed (this may become a thing in the future?)

Reply:
This check was added for perf. As you can see one line below we create an "adaptiveSampler" and then might just drop if we the probe id already exists in `_samplers`..
Once we will decide to update the sampling rate we will need to change `SetRate` anyways

Reply:
We both agreed to leave it like this for the time being and just add inline comment explaining this.
Resolving.